aerleon/lib/yaml.py

```diff
@@ -140,37 +137,130 @@ def ParsePolicy(
         return
 
     # Malformed policy files should generate a PolicyTypeError (unless this is an include file)
-    if 'filters' not in policy_dict or not isinstance(policy_dict['filters'], list):
-
-        if 'terms' in policy_dict:
-            # In this case we are looking at an include file and need to quietly ignore it.
-            return
-
+    if 'filters' in policy_dict and isinstance(policy_dict['filters'], list):
+        pass  # Normal case.
+    elif (
+        depth < MAX_INCLUDE_DEPTH
+        and 'filters_include_only' in policy_dict
+        and isinstance(policy_dict['filters_include_only'], list)
+    ):
+        # Policy files with filters_include_only: are ignored by ParsePolicy but can be included.
+        policy_dict['filters'] = policy_dict['filters_include_only']
+        del policy_dict['filters_include_only']
+    elif 'terms' in policy_dict or 'filters_include_only' in policy_dict:
+        # We are looking at an include file outside of an include and should quietly ignore it.
+        return
+    else:
         raise PolicyTypeError(
             UserMessage("Policy file must contain one or more filter sections.", filename=filename)
         )
 
+    found_filters = []
+
     for filter in policy_dict['filters']:
         # Malformed filters should generate a PolicyTypeError
         if not isinstance(filter, dict):
             raise PolicyTypeError(UserMessage("Filter must be a mapping.", filename=filename))
+
+        def expand_filter(filter):
+            stack = debug_stack.copy()
+            stack.append((filter['__filename__'], filter['__line__']))
+
+            if depth <= 0:
+                raise ExcessiveRecursionError(
+                    UserMessage(
+                        f"Excessive recursion: include depth limit of {MAX_INCLUDE_DEPTH} reached.",  # noqa: E501
+                        filename=filter['__filename__'],
+                        line=filter['__line__'],
+                        include_chain=stack,
+                    )
+                )
+            if not suffix_is_yaml(filter['include']):
+                raise PolicyTypeError(
+                    UserMessage(
+                        f"Policy include source {filter['include']} must end in \".yaml\".",  # noqa: E501
+                        filename=filter['__filename__'],
+                        line=filter['__line__'],
+                        include_chain=stack,
+                    )
+                )
+            include_path = pathlib.Path(base_dir).joinpath(filter['include'])
+            if not _SubpathOf(base_dir, include_path):
+                raise BadIncludePath(
+                    f"Include file cannot be loaded from outside the base directory. File={include_path} base_directory={base_dir}"
+                )
+            try:
+                include_file = _LoadIncludeFile(include_path)
+                include_data = yaml.load(
```

Comment:
Why not use safe_load instead?

Reply:
The param on the next line `Loader=SpanSafeYamlLoader` causes the load to happen in safe mode.